pydantic/_internal/_generate_schema.py

```diff
@@ -1891,20 +1891,21 @@ def _unsubstituted_typevar_schema(self, typevar: typing.TypeVar) -> core_schema.
 
         bound = typevar.__bound__
         constraints = typevar.__constraints__
+        typevar_default_value = getattr(typevar, '__default__', None)
```

Comment:
```suggestion
        default = getattr(typevar, '__default__', None)
```

Reply:
(and below)

Reply:
Is it possible `typevar.__default__` could be `None`? Do we need to use a different sentinel here?

Reply:
Yes, that's why `typing.NoDefault` was introduced. However, this is not an issue anymore with the implementation I suggested below